labkey/utils.py

```diff
@@ -31,3 +31,45 @@ def default(self, o):
 def json_dumps(*args, **kwargs):
     kwargs.setdefault("cls", DateTimeEncoder)
     return json.dumps(*args, **kwargs)
+
+
+def transform_helper(user_transform_func, file_path_run_properties):
+    #file_path_run_properties must be explicitly defined as ${runInfo} within the user's transform script
+    #parse run properties to for results data in and out filepaths
+    file_path_in = ''
+    file_path_out = ''
+    file_run_properties = open(file_path_run_properties) 
+    for l in file_run_properties:
+        row = l.strip().split('\t')
+        if row[0] == 'runDataFile':
+            file_path_out = row[3]
+        if row[0] == 'runDataUploadedFile':
+            file_path_in = row[1]
+    file_run_properties.close()
```

Comment:
```suggestion
    with open(file_path_run_properties) as file_run_properties:
        for l in file_run_properties:
            row = l.strip().split('\t')
            if row[0] == 'runDataFile':
                file_path_out = row[3]
            if row[0] == 'runDataUploadedFile':
                file_path_in = row[1]
```

Reply:
When using open it's best to use it with a `with` statement, that way if any errors happen before close the file handle automatically gets closed.

Reply:
got it, makes sense. in retrospect, I've only rarely seen files opened without a with statement.